### PR TITLE
Make health check fail on timeout or http error

### PR DESCRIPTION
**Pull request by rusch95**
Added a timeout for the http.Get. I also added a check for if there is an error, which might be unnecessary, so remove at your call. 

Testing:

Set timeout to 1 nanoseconds; servers all report failed correctly. Set timeout to 3 seconds, all servers healthy.

### CODE REVIEW

File: health/health.go [hunk 1]
```diff
@@ -86,15 +86,21 @@ func check(
 
 // health checks the given health service
 func health(healthService string) bool {
-	resp, _ := http.Get(healthService)
-	bytes, _ := ioutil.ReadAll(resp.Body)
+	const timeout = time.Duration(3 * time.Second)
```
Comment (cryslith) [inline health/health.go]: I think we should make `timeout` configurable.

File: health/health.go [hunk 2]
```diff
@@ -86,15 +86,21 @@ func check(
...
 
-	resp.Body.Close()
+	client := http.Client{
+		Timeout: timeout,
+	}
 
-	if resp == nil {
+	resp, err := client.Get(healthService)
+	// Check if timeout or HTTP error
+	if resp == nil && err != nil {
```
Comment (cryslith) [inline health/health.go]: This should check only `err != nil`.

File: health/health.go [hunk 3]
```diff
@@ -86,15 +86,21 @@ func check(
...
 		return false
 	}
 
+	bytes, _ := ioutil.ReadAll(resp.Body)
```
Comment (cryslith) [inline health/health.go]: Why ignore errors from reading the response body?

**Comment by rusch95** [comment]
Should I use the timeout passed into Check?

**Comment by cryslith** [comment]
No, that is a different timeout (length of time without receiving any good replies until we declare the backend to be down).  Probably, we should rename them to be less ambiguous.

File: health/health.go [hunk 4]
```diff
@@ -85,15 +88,21 @@ func check(
...
+	resp, err := client.Get(healthService)
+	// Check if response timeouts or returns an HTTP error
+	if err != nil {
 		return false
 	}
+	bytes, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return false
+	}
+	resp.Body.Close()
```
Comment (cryslith) [inline health/health.go:105]: If there is an error in reading the response body, then the body will not be closed.  Probably we should just `defer resp.Body.Close()` before we try to read the body.

**Pull request merged**